### PR TITLE
Mention gccemacs in lsp-doctor and in performance guide

### PR DESCRIPTION
**Pull request by yyoncho**
----

#

### CODE REVIEW

**Review by ericdallo — APPROVED** [review]
Nice :clap:

File: docs/page/performance.md [hunk 1]
```diff
@@ -40,11 +40,14 @@ Benchmarks show that Emacs 27 is `~15 times` faster than Emacs when using Elisp
 ``` elisp
 (setq lsp-idle-delay 0.500)
 ```
+### gccemacs
+
+[gccemacs](https://akrl.sdf.org/gccemacs.html) is the bleading edge version of `Emacs` which byte-compiles elisp to native code resulting in more than 200% speedup. For everyone looking for optimal performance `gccemacs` is the way to go.
```
Comment (leungbk) [inline docs/page/performance.md:45]:
```suggestion
[gccemacs](https://akrl.sdf.org/gccemacs.html) is a bleeding-edge version of `Emacs` that compiles elisp to native code, resulting in more than 200% speedup. For everyone looking for optimal performance, `gccemacs` is the way to go.
```

**Review by leungbk — CHANGES_REQUESTED** [review]

File: lsp-mode.el [hunk 2]
```diff
@@ -7756,10 +7756,12 @@ This avoids overloading the server with many files when starting Emacs."
...
          ,@(-map (-lambda ((msg form))
                    `(insert (format "%s: %s\n" ,msg
-                                    (if (with-current-buffer buf
-                                          ,form)
-                                        (propertize "OK" 'face 'success)
-                                      (propertize "ERROR" 'face 'error)))))
+                                    (let ((res (with-current-buffer buf
+                                                 ,form)))
+                                      (cond
+                                       ((eq res :optional) (propertize "OPTIONAL" 'face 'warning))
```
Comment (kiennq) [inline lsp-mode.el:7762]: `OPTIONAL` is a bit confusing, it doesn't indicate current validity of this feature. Perhaps this should be `NOT AVAILABLE` or `NOT AVAILABLE (OPTIONAL)`?

Comment (yyoncho) [reply lsp-mode.el]: done

**Review by kiennq — APPROVED** [review]

**Review by leungbk — APPROVED** [review]

**Pull request merged**